eng/scripts/Swagger-APIView.ps1

```diff
@@ -0,0 +1,170 @@
+        $readmeFile = Get-ChildItem -Path $currentPath -Filter "readme.md" -File -ErrorAction SilentlyContinue
+        if ($readmeFile -and $readmeFile.Name -eq "readme.md") {
+            return $readmeFile.FullName
+        }
+    }
+
+    return $null
+}
+
+<#
+.DESCRIPTION
+  Use the directory structure convention to get the resource provider name.
+
+.PARAMETER ReadMeFilePath
+  ReadMe File Path for a resource provider.
+
+.OUTPUTS
+  The resource provider name.
+#>
+function Get-ResourceProviderFromReadMePath {
+    param (
+        [Parameter(Mandatory = $true)]
+        [string]$ReadMeFilePath
+    )
+
+    $directoryPath = [System.IO.Path]::GetDirectoryName($ReadMeFilePath)
+    $resourceProviderDirectory = Get-ChildItem -Path $directoryPath -Directory | Select-Object -First 1
+    return $resourceProviderDirectory.Name
+}
+
+<#
+.DESCRIPTION
+  Invoke the swagger parset to generate APIView tokens.
+
+.PARAMETER Type
+  New or Baseline swagger APIView tokens.
+
+.PARAMETER ReadMeFilePath
+  The Swagger ReadMeFilePath.
+
+.PARAMETER ResourceProvider
+  The ResourceProvider Name.
+
+.PARAMETER Tag
+  The Tag to use for generating the APIView Tokens.
+
+.PARAMETER TokenDirectory
+  The directory to store the generated APIView Tokens.
+
+.OUTPUTS
+  The resource provider name.
+#>
+function Invoke-SwaggerAPIViewParser {
+    param (
+        [ValidateSet("New", "Baseline")]
+        [Parameter(Mandatory = $true)]
+        [string]$Type,
+        [Parameter(Mandatory = $true)]
+        [string]$ReadMeFilePath,
+        [Parameter(Mandatory = $true)]
+        [string]$ResourceProvider,
+        [Parameter(Mandatory = $true)]
+        [string]$TokenDirectory,
+        [string]$Tag
+    )
+    $tempWorkingDirectoryName = [guid]::NewGuid().ToString()
+    $tempWorkingDirectoryPath = [System.IO.Path]::Combine($TempDirectory, $tempWorkingDirectoryName)
+    New-Item -ItemType Directory -Path $tempWorkingDirectoryPath
+
+    Set-Location -Path $tempWorkingDirectoryPath
+
+    try {
+      # Generate Swagger APIView tokens
+      $command = "& dotnet swaggerAPIParser --readme $readMeFile --package-name $resourceProvider --use-tag-for-output"
+
+      if ($Tag) {
+          $command += " --tag $Tag"
+      }
+        
+      Invoke-Expression $command
+      $generatedAPIViewTokenFile = Get-ChildItem -Path $tempWorkingDirectoryPath -File | Select-Object -First 1
+      $readMeTag = $generatedAPIViewTokenFile.BaseName
+
+      Write-Host "${green}Generated '$Type' APIView Token File using file, '$readMeFile' and tag '$readMeTag'${reset}"
+
+      $apiViewTokensFilePath = [System.IO.Path]::Combine($TokenDirectory, "$resourceProvider.$Type.json")
+      Move-Item -Path $generatedAPIViewTokenFile.FullName -Destination $apiViewTokensFilePath -Force
+      return $readMeTag
+    } catch {
+      Write-Host "${red}Failed to generate '$Type' APIView Tokens using '$readMeFile' for '$resourceProvider'${reset}"
```

Comment:
Instead of outputting color here why not use the logging functions and log an error? On the shell it will show red and in devops it will correctly cause a failure to happen.